lib/Doctrine/ORM/Persisters/Collection/OneToManyPersister.php

```diff
@@ -163,7 +164,20 @@ public function removeElement(PersistentCollection $collection, $element)
         $mapping   = $collection->getMapping();
         $persister = $this->uow->getEntityPersister($mapping['targetEntity']);
 
-        return $persister->delete($element);
```

Comment:
I think this was intended to be used only be used with `orphanRemoval=true`, not always like it is now.